nixos/modules/system/boot/systemd/sysusers.nix

```diff
@@ -5,6 +5,8 @@ let
   cfg = config.systemd.sysusers;
   userCfg = config.users;
 
+  systemUsers = lib.filterAttrs (_username: opts: !opts.isNormalUser) userCfg.users;
```

Comment:
why not filter for isSystemUser?

Reply:
Filtering for isSystemUser is actually not enough because we only assert for effective system users (which also includes users with a uid < 1000). `!isNormalUser` is thus the most reliable way to figure out if a user is a system user.

Reply:
What does it mean if a user is not a normal user and not a system user?